src/librustc_metadata/rmeta/encoder.rs

```diff
@@ -614,6 +607,11 @@ impl EncodeContext<'tcx> {
         self.encode_promoted_mir(def_id);
     }
 
+    fn encode_def_and_ident_spans(&mut self, def_id: DefId, ident_span: Span) {
+        record!(self.per_def.span[def_id] <- self.tcx.def_span(def_id));
+        self.encode_ident_span(def_id, ident_span);
```

Comment:
It's a bit strange to pair these two unrelated things together into a single function.

Reply:
I consider them related - we're encoding two different spans for the same item. However, this method is less useful now that we no longer looks things up from an intermediate table.